name: gfl-data-miner auto update bot

on:
  workflow_dispatch:
    inputs:
      cmdargs:
        description: 'Commandline Arguments'     
        required: false
        default: 'tw kr jp us ch' 
        type: string
  schedule:
    - cron: '0 0/3 * * *'
    - cron: '30 1-13 * * 4'
  push:
    paths:
      - 'data-miner'
    

jobs:
  gfl-data-miner-update:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
        with:
          token: ${{ secrets.PERSONAL_TOKEN }}
          submodules: 'recursive'

      - run: git submodule foreach --recursive git checkout main
      - run: git config --global core.autocrlf input

      - name: Setup Python
        uses: actions/setup-python@v2.2.2
        with:
          python-version: '3.8'
          
      - name: Install Data-Miner Dependencies
        uses: py-actions/py-dependency-install@v3
        with:
          path: "data-miner/requirements.txt"
      - name: Data Miner
        id: data-miner
        run: | 
          python data-miner/data_miner.py ${{ inputs.cmdargs || 'tw kr jp us ch at' }} ${{ github.event_name=='push' && '-f' || '' }}
      
      - name: Commit-tw
        continue-on-error: true
        uses: ./.github/actions/DataMinerCommit
        with:
          commit_message: ${{ steps.data-miner.outputs.commit-message-tw }}
          repository: data/tw
      - name: Commit-kr
        continue-on-error: true
        uses: ./.github/actions/DataMinerCommit
        with:
          commit_message: ${{ steps.data-miner.outputs.commit-message-kr }}
          repository: data/kr
      - name: Commit-jp
        continue-on-error: true
        uses: ./.github/actions/DataMinerCommit
        with:
          commit_message: ${{ steps.data-miner.outputs.commit-message-jp }}
          repository: data/jp
      - name: Commit-us
        continue-on-error: true
        uses: ./.github/actions/DataMinerCommit
        with:
          commit_message: ${{ steps.data-miner.outputs.commit-message-us }}
          repository: data/us
      - name: Commit-ch
        continue-on-error: true
        uses: ./.github/actions/DataMinerCommit
        with:
          commit_message: ${{ steps.data-miner.outputs.commit-message-ch }}
          repository: data/ch
      - name: Commit-at
        continue-on-error: true
        uses: ./.github/actions/DataMinerCommit
        with:
          commit_message: ${{ steps.data-miner.outputs.commit-message-at }}
          repository: data/at
      - name: Push
        run: git push